### PR TITLE
dnn: add another color to Net::Impl::dump()

### LINKED ISSUE #19034: dnn: buffer overflow in Net::Impl::dump()
##### System information (version)

- OpenCV => 4.5.1-pre
- Operating System / Platform => Windows 64 Bit
- Compiler => mingw64
##### Detailed description

only 8 colors here:
https://github.com/opencv/opencv/blob/31619faa70b889bf45bec2e9b6139fb4fd2b8238/modules/dnn/src/dnn.cpp#L4464

but 9 colorId's used here:
https://github.com/opencv/opencv/blob/31619faa70b889bf45bec2e9b6139fb4fd2b8238/modules/dnn/src/dnn.cpp#L4609

it's only in the master branch, 3.4 has it correct (6 colors / colorId's)

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [ ] There is reproducer code and related data files: videos, images, onnx, etc

### PR DESCRIPTION
**Pull request by berak**
resolves #19034

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/src/dnn.cpp [hunk 1]
```diff
@@ -4459,7 +4459,7 @@ string Net::Impl::dump()
             prevNode = itBackend->second;
         }
     }
-    string colors[] = {"#ffffb3", "#fccde5", "#8dd3c7", "#bebada", "#80b1d3", "#fdb462", "#ff4848", "#b35151"};
+    string colors[] = {"#ffffb3", "#fccde5", "#8dd3c7", "#bebada", "#80b1d3", "#fdb462", "#ff4848", "#b35151", "#b662ff"};
```
Comment (dkurt) [inline modules/dnn/src/dnn.cpp:4462]: Does it make sense to replace to `std::vector<std::string>` and check `colorId < colors.size()`?

Comment (berak) [reply modules/dnn/src/dnn.cpp:4462]: btw, cppcheck found this here.

arent there static analyzers running over the code on the buildbots ?

Comment (asmorkalov) [reply modules/dnn/src/dnn.cpp:4462]: It's done time to time, but not for each PR.

Comment (asmorkalov) [reply modules/dnn/src/dnn.cpp:4462]: The check is good idea!

Comment (berak) [reply modules/dnn/src/dnn.cpp:4462]: so i'll add it ...

File: modules/dnn/src/dnn.cpp [hunk 2]
```diff
@@ -4613,6 +4613,7 @@ string Net::Impl::dump()
             case DNN_TARGET_CUDA_FP16: out << "CUDA_FP16"; colorId = 6; break;
             // don't use default:
         }
+        CV_Assert(colorId < colors.size());
```
Comment (alalek) [inline modules/dnn/src/dnn.cpp:4616]: BTW, this check can help in runtime only.
(no problem detection if there is no tests for this code)

**Review by alalek — APPROVED** [review]
Thank you :+1:

**Review by dkurt — APPROVED** [review]

File: modules/dnn/src/dnn.cpp (hunk 2 above)
Comment (berak) [reply modules/dnn/src/dnn.cpp:4616]: yes, i'd also rather favour something, that throws a compile time error instead.

unfortunately, all i could come up with is this:

    enum X {A,B,C,D,MAX_X};
    std::array<string,MAX_X> {"A","B","C","D","E"};

(works if there are excessive strings, but stays silent if there are not enough)

Comment (alalek) [reply modules/dnn/src/dnn.cpp:4616]: If you want to add "MAX" value into enum then please use this approach to hide that from documentation:

https://github.com/opencv/opencv/blob/4.5.0/modules/calib3d/include/opencv2/calib3d.hpp#L474-L476

**Pull request merged**